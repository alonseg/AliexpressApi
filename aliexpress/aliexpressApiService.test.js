const AliexpressApiService = require("./aliexpressApiService");
const {TopClient} = require('./lib/api/topClient');


jest.mock('./lib/api/topClient');

let executePromisifiedMock = jest.fn();
let getPromisifiedMock = jest.fn();
let getMock = jest.fn();
let executeMock = jest.fn();

describe('aliexpressApiService', () => {
    beforeAll(()=>{
        TopClient.mockImplementation(()=>{
                return {
                    executePromisified: executePromisifiedMock,
                    getPromisified: getPromisifiedMock,
                    get: getMock,
                    execute: executeMock
                }
        });
    })
    beforeEach(() => {
        executePromisifiedMock.mockClear();
        getPromisifiedMock.mockClear();
        getMock.mockClear();
        executeMock.mockClear();
    });
    
    describe('getAffiliateLink', () => {
        it('should call getPromisified with the correct params', () => {
            const aliexpress = new AliexpressApiService('key', 'secret', 'signature', 'trackingId');
            aliexpress.client.getPromisified = getPromisifiedMock;
            aliexpress.getAffiliateLink('www.test.com');
            expect(getPromisifiedMock.mock.calls).toMatchSnapshot();
        });
    });

    describe('getAffiliateLinks', () => {
        it('should call getPromisified with the correct params', () => {
            const aliexpress = new AliexpressApiService('key', 'secret', 'signature', 'trackingId');
            aliexpress.client.getPromisified = getPromisifiedMock;
            aliexpress.getAffiliateLink(['www.test.com', 'www.test2.com']);
            expect(getPromisifiedMock.mock.calls).toMatchSnapshot();
        });
    });

    describe('getProductsDetails', () => {
        it('should call getPromisified with the correct params', () => {
            const aliexpress = new AliexpressApiService('key', 'secret', 'signature', 'trackingId');
            aliexpress.client.getPromisified = getPromisifiedMock;
            aliexpress.getProductsDetails(['12334', '4321']);
            expect(getPromisifiedMock.mock.calls).toMatchSnapshot();
        });
    });
});