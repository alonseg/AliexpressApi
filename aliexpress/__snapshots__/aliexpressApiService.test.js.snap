// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aliexpressApiService getAffiliateLink should call getPromisified with the correct params 1`] = `
Array [
  Array [
    "aliexpress.affiliate.link.generate",
    Object {
      "app_signature": "signature",
      "promotion_link_type": "0",
      "source_values": "www.test.com",
      "tracking_id": "trackingId",
    },
  ],
]
`;

exports[`aliexpressApiService getAffiliateLinks should call getPromisified with the correct params 1`] = `
Array [
  Array [
    "aliexpress.affiliate.link.generate",
    Object {
      "app_signature": "signature",
      "promotion_link_type": "0",
      "source_values": "www.test.com,www.test2.com",
      "tracking_id": "trackingId",
    },
  ],
]
`;

exports[`aliexpressApiService getProductsDetails should call getPromisified with the correct params 1`] = `
Array [
  Array [
    "aliexpress.affiliate.productdetail.get",
    Object {
      "app_signature": "signature",
      "country": "US",
      "fields": "",
      "product_ids": "12334,4321",
      "target_currency": "USD",
      "target_language": "EN",
      "tracking_id": "trackingId",
    },
  ],
]
`;
